paddle/inference/tests/book/test_inference_recognize_digits.cc

```diff
@@ -58,6 +58,47 @@ void TestInference(const std::string& dirname,
   delete scope;
 }
 
+template <typename T>
```

Comment:
Could `SetupTensor` and `CheckError` be public functions, which not only used for test_inference_recognize_digits? These two functions seem to be used in all unit tests.

Reply:
Yes. #8058 the definition of `TestInference` is moved into a separate header file `test_helper.h`. To avoid conflict changes, we can merge this PR first and move `SetupTensor` and `CheckError` in #8058 or other PR.

Reply:
I will fix this SetupTensor and CheckError issue in #8058